sycl/test-e2e/ESIMD/unified_memory_api/Inputs/gather.hpp

```diff
@@ -761,3 +761,254 @@ template <typename T, TestFeatures Features> bool testACC(queue Q) {
   }
   return Passed;
 }
+
+template <typename T, uint16_t N, uint16_t VS, bool UseMask, bool UsePassThru,
+          bool UseProperties, typename PropertiesT>
+bool testLACC(queue Q, uint32_t MaskStride, PropertiesT) {
+
+  static_assert(VS > 0 && N % VS == 0,
+                "Incorrect VS parameter. N must be divisible by VS.");
+  constexpr int NOffsets = N / VS;
+  static_assert(!UsePassThru || UseMask,
+                "PassThru cannot be used without using mask");
+
+  constexpr uint32_t Groups = 8;
+  constexpr uint32_t Threads = 16;
+
+  std::cout << "Running case: T=" << esimd_test::type_name<T>() << ", N=" << N
+            << ", VS=" << VS << ", MaskStride=" << MaskStride
+            << ", Groups=" << Groups << ", Threads=" << Threads
+            << ", use_mask=" << UseMask << ", use_pass_thru=" << UsePassThru
+            << ", use_properties=" << UseProperties << std::endl;
+
+  uint16_t Size = Groups * Threads * N;
+  using Tuint = esimd_test::uint_type_t<sizeof(T)>;
+
+  sycl::range<1> GlobalRange{Groups};
+  sycl::range<1> LocalRange{Threads};
+  sycl::nd_range<1> Range{GlobalRange * LocalRange, LocalRange};
+
+  T *Out = sycl::malloc_shared<T>(Size, Q);
+  std::memset(Out, 0, Size * sizeof(T));
+
+  T *In = sycl::malloc_shared<T>(Size * 2, Q);
+  for (int I = 0; I < Size; I++)
+    In[I] = esimd_test::getRandomValue<T>();
+
+  try {
+    buffer<T, 1> InBuf(In, Size * 2);
+    Q.submit([&](handler &CGH) {
+       // Allocate a bit more to safely initialize it with 8-element chunks.
+       constexpr uint32_t SLMSize = (Threads * N + 8) * sizeof(T);
+
+       auto InAcc = local_accessor<T, 1>(SLMSize, CGH);
+
+       CGH.parallel_for(Range, [=](sycl::nd_item<1> NDI) SYCL_ESIMD_KERNEL {
+         uint16_t GlobalID = NDI.get_global_id(0);
+         uint16_t LocalID = NDI.get_local_id(0);
+         uint32_t GlobalElemOffset = GlobalID * N;
+         uint32_t LocalElemOffset = LocalID * N;
+
+         if (LocalID == 0) {
+           for (int I = 0; I < Threads * N; I += 8) {
+             simd<T, 8> InVec(In + GlobalElemOffset + I);
+             simd<uint32_t, 8> offsets(I * sizeof(T), sizeof(T));
```

Comment:
nit: if use `SomeVar` style, then
```suggestion
             simd<uint32_t, 8> Offsets(I * sizeof(T), sizeof(T));
```

Reply:
+1 more comment: Please ensure there are calls and compile-tests like: 
gather(lacc, byte_offsets, 0), and gather(lacc, byte_offsets, 0, mask)
to verify lacc-ga-4,5  having MaskT template parameter is sufficient work-around for 2 older functions having 'glob_offsets' parameter.